conda_pack/tests/test_core.py

```diff
@@ -255,7 +255,26 @@ def test_roundtrip(tmpdir, py37_env):
             assert not member.startswith(os.path.sep)
 
         extract_path = str(tmpdir.join('env'))
-        fil.extractall(extract_path)
+        def is_within_directory(directory, target):
+            
+            abs_directory = os.path.abspath(directory)
+            abs_target = os.path.abspath(target)
+        
+            prefix = os.path.commonprefix([abs_directory, abs_target])
+            
+            return prefix == abs_directory
+        
+        def safe_extract(tar, path=".", members=None, *, numeric_owner=False):
+        
+            for member in tar.getmembers():
+                member_path = os.path.join(path, member.name)
+                if not is_within_directory(path, member_path):
+                    raise Exception("Attempted Path Traversal in Tar File")
+        
+            tar.extractall(path, members, numeric_owner=numeric_owner) 
```

Comment:
https://github.com/conda/conda-package-streaming/blob/main/conda_package_streaming/extract.py#L38 passes an iterator to tar to avoid seeking backwards in the tarfile. This is important to avoid decompressing the tarball twice, once to iterate over getmembers() and once to extractall(). Is it necessary to call os.path.abspath() and os.path.commonprefix() for every file?